From ebeb4e6f47d7570b35e03a12e642a74c926c29b9 Mon Sep 17 00:00:00 2001
From: Jason King <jason.king@joyent.com>
Date: Tue, 29 May 2018 09:05:49 -0500
Subject: [PATCH] OS-6980 libvarpd leaks varpd_query_ts Reviewed by: Robert
 Mustacchi <rm@joyent.com> Reviewed by: Dan McDonald <danmcd@joyent.com>
 Approved by: Dan McDonald <danmcd@joyent.com>

---
 usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c b/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c
index b7506c56cf..124e3c5791 100644
--- a/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c
+++ b/usr/src/lib/varpd/libvarpd/common/libvarpd_overlay.c
@@ -514,6 +514,8 @@ libvarpd_plugin_query_reply(varpd_query_handle_t *vqh, int action)
 	else
 		libvarpd_panic("plugin %s passed in an invalid action: %d",
 		    vqp->vq_instance->vri_plugin->vpp_name, action);
+
+	umem_cache_free(vqp->vq_instance->vri_impl->vdi_qcache, vqp);
 }
 
 void
-- 
2.21.0

